### PR TITLE
Add promql funcs hour,minute,month,year,day_of_month,day_of_week,day_of_year,days_in_month

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Added promql functions hour(),minute(),month(),year(),day_of_month(),day_of_week(),day_of_year(),days_in_month()

# Testing
Tested the above functions using query API. 
Added testcases for all the above functions
# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/integrations/prometheus/promql/metricsconsts.go
Comment (Jiale-Fang) [inline pkg/integrations/prometheus/promql/metricsconsts.go:243]: function name should be: `day_of_month`, you should modify other funcs as well

**Review by Jiale-Fang — CHANGES_REQUESTED** [review]

**Review by nkunal — COMMENTED** [review]
lgtm, once @AndrewHess approves, it can be merged

File: pkg/segment/results/mresults/seriesresult.go
Comment (AndrewHess) [inline pkg/segment/results/mresults/seriesresult.go:817]: nit: take this one step further to reduce duplicate code; do `t := time.Unix(int64(dpTs), 0).UTC()` inside `evaluateTimeFunc()` instead of having that line every time you call `evaluateTimeFunc()`

File: cicd/metrics_test.csv
Comment (venkateshamatam) [reply cicd/metrics_test.csv:50]: yes, day_of_week returns a value between 0(Sunday) and 6 and day_of_year returns a value between 1 and 366.

File: pkg/integrations/prometheus/promql/metricsconsts.go
Comment (AndrewHess) [inline pkg/integrations/prometheus/promql/metricsconsts.go:253]: This discription doesn't match promQL: https://prometheus.io/docs/prometheus/latest/querying/functions/#day_of_week
It should be 0 to 6 inclusive, with 0=Sunday, 1=Monday, ...

Comment (venkateshamatam) [reply pkg/integrations/prometheus/promql/metricsconsts.go]: I just updated it

**Review by AndrewHess — DISMISSED** [review]

**Review by AndrewHess — DISMISSED** [review]

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**
